tests/callbacks/test_curriculum_learning_callback.py

```diff
@@ -237,8 +239,11 @@ def test_curriculum_learning_callback_state_dict(build_tiny_mpt: Callable,):
     callback.iteration_start(state, logger)
     callback.iteration_end(state, logger)
     assert callback.state_dict() == {
-        'schedule': kwargs['schedule'],
-        'schedule_index': 1,
+        'state':
```

Comment:
jw, did these tests catch the state dict issues though? Is there a way we can actually test that this has fixed the state dict flattening issue?

Reply:
Unfortunately not, Chuck is working on a longer term fix

Reply:
Added